setup.py

```diff
@@ -34,17 +34,25 @@
 # ============================================================================
 #
 import setuptools
+from pathlib import Path, PurePath
 
 with open("README.md", "r") as file:
 	long_description = file.read()
 
+projectName = "pyVHDLParser"
+
+rfile = Path("requirements.txt")

```

Comment:
Using Pathlib is even much simpler :).

```diff
-from pathlib import Path, PurePath
+from pathlib import Path

projectName = "pyVHDLParser"

rfile = Path("requirements.txt")
-root = PurePath(__file__)
-if not Path.is_file(rfile):
-  rfile = Path(root.parent, projectName + ".egg-info", "requires.txt")
-  if not Path.is_file(rfile):
+root = Path(__file__)
+if not rfile.is_file():
+  rfile = root.parent / (projectName + ".egg-info") / "requires.txt"
+  if not rfile.is_file():
		exit(1)
requirements = []
-with open(rfile) as file:
+with rfile.open() as file:
```

Pathlib is fully OO.
* They overloaded the `/` operator for path concatenation :).
* `is_file` is a method, not a classmethod -> data knows best how to test itself.

Reply:
Thanks a lot! I updated the PR accordingly...